urnx/parse.go

```diff
@@ -0,0 +1,29 @@
+package urnx
+
+import (
+	"strings"
+
+	"github.com/google/uuid"
+)
+
+func Parse(urn string) (*URN, error) {
+	conv := strings.Split(urn, ":")
+
+	if conv[0] != PREFIX {
+		return nil, ErrInvalidURNPrefix
+	}
+
+	if len(conv) != URNLENGTH {
+		return nil, ErrInvalidURN
+	}
+
+	u := &URN{
+		Prefix:       conv[0],
+		Namespace:    conv[1],
+		ResourceType: conv[2],
```

Comment:
Do we want to place any restrictions on what format a resource type can have? Presumably alphabetical only with `-` also supported maybe?

Reply:
I would say at least alphanumeric and `-` ? Thinking along the lines of what if something like `s3-bucket` were an option